packages/near-membrane-base/src/membrane.ts

```diff
@@ -114,1367 +123,1424 @@ export type DistortionCallback = (target: ProxyTarget) => ProxyTarget;
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        // __lookupSetter__: ObjectProto__lookupSetter__,
-        hasOwnProperty: ObjectProtoHasOwnProperty,
-    } = Object.prototype as any;
-    const proxyTargetToPointerMap = new WeakMap();
-    const LockerLiveValueMarkerSymbol = Symbol.for('@@lockerLiveValue');
-    const distortionCallback = optionalDistortionCallback || ((o) => o);
-
-    let selectedTarget: undefined | ProxyTarget;
-    let foreignCallablePushTarget: CallablePushTarget;
-    let foreignCallableApply: CallableApply;
-    let foreignCallableConstruct: CallableConstruct;
-    let foreignCallableDefineProperty: CallableDefineProperty;
-    let foreignCallableDeleteProperty: CallableDeleteProperty;
-    let foreignCallableGetOwnPropertyDescriptor: CallableGetOwnPropertyDescriptor;
-    let foreignCallableGetPrototypeOf: CallableGetPrototypeOf;
-    let foreignCallableHas: CallableHas;
-    let foreignCallableIsExtensible: CallableIsExtensible;
-    let foreignCallableOwnKeys: CallableOwnKeys;
-    let foreignCallablePreventExtensions: CallablePreventExtensions;
-    let foreignCallableSetPrototypeOf: CallableSetPrototypeOf;
-    let foreignCallableGetTargetIntegrityTraits: CallableGetTargetIntegrityTraits;
-    let foreignCallableHasOwnProperty: CallableHasOwnProperty;
-
-    // eslint-disable-next-line no-shadow
-    enum TargetTraits {
-        None = 0,
-        IsArray = 1 << 0,
-        IsFunction = 1 << 1,
-        IsObject = 1 << 2,
-        IsArrowFunction = 1 << 3,
-        Revoked = 1 << 4,
-    }
-
-    // eslint-disable-next-line no-shadow
-    enum TargetIntegrityTraits {
-        None = 0,
-        IsNotExtensible = 1 << 0,
-        IsSealed = 1 << 1,
-        IsFrozen = 1 << 2,
-        Revoked = 1 << 4,
-    }
-
-    if (
-        ObjectProtoHasOwnProperty.call(Error, 'stackTraceLimit') &&
-        typeof Error.stackTraceLimit === 'number'
-    ) {
-        // The default stack trace limit is 10. Increasing to 20 as a baby step.
-        Error.stackTraceLimit *= 2;
-    }
-
-    function isUndefinedSymbol(v: any): boolean {
-        return v === undefinedSymbol;
-    }
-
-    // TODO: this is really only needed if Realm constructor is not available, because
-    // it will prevent errors from another realm to leak into the caller realm, but
-    // if Realm constructor is in place, the callable boundary takes care of it.
-    function foreignErrorControl<T extends (...args: any[]) => any>(foreignFn: T): T {
-        return <T>function foreignErrorControlFn(...args: any[]): any {
-            try {
-                return foreignFn(...args);
-            } catch (e) {
-                const pushedError = getSelectedTarget();
-                if (pushedError) {
-                    throw pushedError;
-                }
-                throw new TypeErrorCtor(e.message);
-            }
-        };
-    }
+) => HooksCallback {
+    return function curryInit(
```

Comment:
Currying the init function to bring in instrumentation lib object. Almost all the diff blocks below are bogus. Github diff is confused with the changes. Beyond Compare and VS Code shows the diff fine.